src/Functions/FunctionsConversion.h

```diff
@@ -2825,13 +2825,13 @@ struct ToNumberMonotonicity
         /// (Enum has separate case, because it is different data type)
         if (checkAndGetDataType<DataTypeNumber<T>>(&type) ||
             checkAndGetDataType<DataTypeEnum<T>>(&type))
-            return { .is_monotonic = true, .is_always_monotonic = true };
+            return { .is_monotonic = true, .is_always_monotonic = true, .is_strict = true };
```

Comment:
Doesn't this case include float-to-int, which is not strictly monotonic?

Reply:
No, here only handles cases where to_type and from_type are the same.

Reply:
> No, here only handles cases where to_type and from_type are the same.

No, here only handles cases where to_type and from_type are the same.